### ISSUE TITLE
airframe-config cannot load nested case classes

### ISSUE THREAD

**Issue by 0x1997** [wvlet/airframe#110]
```Scala
final case class ServerConfig(host: String, port: Int)
final case class ServerPoolConfig(servers: IndexedSeq[ServerConfig])

val config = Config(env = configEnv, configPaths = Seq(configPath))
      .registerFromYaml[ServerPoolConfig]("servers.yml")
```
```Yaml
default:
  servers:
    - host: 10.0.0.1
      port: 5000
    - host: 10.0.0.2
      port: 5000
```
```
Exception in thread "main" java.util.NoSuchElementException: key not found: class com.foo.ServerConfig
        at scala.collection.MapLike$class.default(MapLike.scala:228)
        at scala.collection.AbstractMap.default(Map.scala:59)
        at scala.collection.MapLike$class.apply(MapLike.scala:141)
        at scala.collection.AbstractMap.apply(Map.scala:59)
        at wvlet.surface.Primitive$.apply(Surfaces.scala:61)
        at wvlet.surface.reflect.TypeConverter$.convert(TypeConverter.scala:102)
        at wvlet.surface.reflect.TypeConverter$.convert(TypeConverter.scala:75)
        at wvlet.surface.reflect.StandardBuilder$$anonfun$set$1.apply(ObjectBuilder.scala:156)
        at wvlet.surface.reflect.StandardBuilder$$anonfun$set$1.apply(ObjectBuilder.scala:155)
        at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
        at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
        at scala.collection.Iterator$class.foreach(Iterator.scala:893)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
        at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
        at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
        at wvlet.surface.reflect.StandardBuilder$class.set(ObjectBuilder.scala:155)
        at wvlet.surface.reflect.SimpleObjectBuilder.set(ObjectBuilder.scala:213)
        at wvlet.surface.reflect.GenericBuilder$class.set(ObjectBuilder.scala:61)
        at wvlet.surface.reflect.SimpleObjectBuilder.set(ObjectBuilder.scala:213)
        at wvlet.config.YamlReader$$anonfun$bind$2.apply(YamlReader.scala:66)
        at wvlet.config.YamlReader$$anonfun$bind$2.apply(YamlReader.scala:65)
        at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
        at scala.collection.Iterator$class.foreach(Iterator.scala:893)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
        at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
        at wvlet.config.YamlReader$.bind(YamlReader.scala:65)
        at wvlet.config.YamlReader$$anonfun$loadMapOf$2.apply(YamlReader.scala:44)
        at wvlet.config.YamlReader$$anonfun$loadMapOf$2.apply(YamlReader.scala:43)
        at scala.collection.TraversableLike$WithFilter$$anonfun$map$2.apply(TraversableLike.scala:683)
        at scala.collection.immutable.Map$Map2.foreach(Map.scala:137)
        at scala.collection.TraversableLike$WithFilter.map(TraversableLike.scala:682)
        at wvlet.config.YamlReader$.loadMapOf(YamlReader.scala:43)
        at wvlet.config.Config.loadFromYaml(Config.scala:177)
        at wvlet.config.Config.registerFromYaml(Config.scala:160)
        ...
```

**Comment by xerial** [comment]
Thanks for reporting. It seems there is a bug in finding nested class Surface.

**Comment by xerial** [comment]
@0x1997 I'm now implementing general purpose object serializer/deserializer in #107. I'll address this issue after #107 is done.

**Comment by 0x1997** [comment]
Great! Thanks.

**Comment by xerial** [comment]
@0x1997 Fixed this issue in airframe 0.27. For JavaBean mapping #109, it would be possible to add JavaBeanCodec (mapping from javabean -> msgpack -> javabean) in airframe-tablet. 

For customizing mappings (e.g., string value -> some Java/Scala classes), we need to think about how to pass such custom mapping to Config.
